python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py

```diff
@@ -17,19 +19,52 @@ def test_basic_resource_request(
     resource = FivetranWorkspace(
         account_id=TEST_ACCOUNT_ID, api_key=TEST_API_KEY, api_secret=TEST_API_SECRET
     )
-
     client = resource.get_client()
-    client.get_connector_details(connector_id=connector_id)
+
+    # fetch workspace data calls
     client.get_connectors_for_group(group_id=group_id)
     client.get_destination_details(destination_id=destination_id)
     client.get_groups()
     client.get_schema_config_for_connector(connector_id=connector_id)
 
-    assert len(all_api_mocks.calls) == 5
-
+    assert len(all_api_mocks.calls) == 4
     assert "Basic" in all_api_mocks.calls[0].request.headers["Authorization"]
+    assert group_id in all_api_mocks.calls[0].request.url
+    assert destination_id in all_api_mocks.calls[1].request.url
+    assert "groups" in all_api_mocks.calls[2].request.url
+    assert f"{connector_id}/schemas" in all_api_mocks.calls[3].request.url
+
+    # connector details calls
+    all_api_mocks.calls.reset()
+    client.get_connector_details(connector_id=connector_id)
+    client.update_schedule_type_for_connector(connector_id=connector_id, schedule_type="auto")
+
+    assert len(all_api_mocks.calls) == 2
     assert connector_id in all_api_mocks.calls[0].request.url
-    assert group_id in all_api_mocks.calls[1].request.url
-    assert destination_id in all_api_mocks.calls[2].request.url
-    assert "groups" in all_api_mocks.calls[3].request.url
-    assert f"{connector_id}/schemas" in all_api_mocks.calls[4].request.url
+    assert connector_id in all_api_mocks.calls[1].request.url
+    assert all_api_mocks.calls[1].request.method == "PATCH"
+
+    # sync calls
+    all_api_mocks.calls.reset()
+    client.start_sync(connector_id=connector_id)
+    assert len(all_api_mocks.calls) == 3
+    assert f"{connector_id}/force" in all_api_mocks.calls[2].request.url
+
+    # resync calls
+    all_api_mocks.calls.reset()
+    client.start_resync(connector_id=connector_id, resync_parameters=None)
+    assert len(all_api_mocks.calls) == 3
+    assert f"{connector_id}/resync" in all_api_mocks.calls[2].request.url
+
+    # resync calls with parameters
+    all_api_mocks.calls.reset()
+    client.start_resync(connector_id=connector_id, resync_parameters={"property1": ["string"]})
+    assert len(all_api_mocks.calls) == 3
+    assert f"{connector_id}/schemas/tables/resync" in all_api_mocks.calls[2].request.url
+
+    # poll calls
+    all_api_mocks.calls.reset()
+    client.poll_sync(
+        connector_id=connector_id, previous_sync_completed_at=parser.parse(MIN_TIME_STR)
+    )
```

Comment:
I'm finding these tests to be pretty insufficient, this is OK maybe for the underlying start_xsync fxns, but sync_and_poll, and resync_and_poll have pretty complicated logic that I think is worth testing the state changes on.

Reply:
I tend to prefer PRs to be relatively atomic in their additions + tests - so I think yea makes sense to add some tests here as well.

For what it's worth, with only 9/n and 10/n open, there was no way for me to know that context. I think it's useful to try to think about things from that perspective when writing up PR description, adding comments etc.

Reply:
That's totally fair - I will update the tests here and make sure to apply this feedback in the future as well.

Reply:
sounds good! Thanks